Drivers/Src/Switches.c

```diff
@@ -1,7 +1,10 @@
 #include "Switches.h"
+#include "os.h"
+#include "Tasks.h"
+#include "BSP_GPIO.h"
 
-
-
+static OS_MUTEX CommMutex; //Mutex to lock SPI lines
+static uint16_t SwitchStates_Bitmap;
```

Comment:
We should initialize this to all zeroes in case we do a read before we do a write